pkg/network/socket_filter.go

```diff
@@ -0,0 +1,31 @@
+// +build linux_bpf
+
+package network
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/ebpf/manager"
+)
+
+// HeadlessSocketFilter creates a raw socket attached to the given socket filter.
+// The underlying raw socket isn't polled and the filter is not meant to accept any packets.
+// The purpose is to use this for pure eBPF packet inspection.
+// TODO: After the proof-of-concept we might want to replace the SOCKET_FILTER program by a TC classifier
+func HeadlessSocketFilter(rootPath string, filter *manager.Probe) (closeFn func(), err error) {
+	var (
+		packetSrc *packetSource
+		srcErr    error
+	)
+
+	nsErr := util.WithRootNS(rootPath, func() {
+		packetSrc, srcErr = newPacketSource(filter)
+	})
+	if nsErr != nil {
+		return nil, nsErr
+	}
+	if srcErr != nil {
+		return nil, srcErr
+	}
```

Comment:
Besides the point of this PR, but this is a bit awkward  to read - we could probably just change the function argument interface so its written as:

```go
err := util.WithRootNS(rootPath, func() err {
	packetSrc, srcErr = newPacketSource(filter)
	return srcErr
})
if err != nil {
	return nil, err
}
```

where `err = nsErr or srcErr` within `util.WithRootNS()`

Reply:
Agreed 👍 I don't mind changing that in this PR

Reply:
Done